go-controller/cmd/ovnkube-identity/ovnkubeidentity.go

```diff
@@ -0,0 +1,377 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"strconv"
+	"sync"
+	"syscall"
+	"time"
+
+	certificatesv1 "k8s.io/api/certificates/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+	utilpointer "k8s.io/utils/pointer"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/certwatcher"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressip/v1/apis/clientset/versioned/scheme"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/csrapprover"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovnwebhook"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/sys/unix"
+)
+
+type config struct {
+	kubeconfig          string
+	apiServer           string
+	logLevel            int
+	port                int
+	host                string
+	certDir             string
+	metricsAddress      string
+	leaseNamespace      string
+	enableInterconnect  bool
+	enableHybridOverlay bool
+	disableWebhook      bool
```

Comment:
This is not required upstream but it might be used to run the webhook and the approver as separate containers.
It is useful for us downstream as when the leader is lost in the approver the whole process has to exit, we don't want the webhook to exit as well.